File: Easy/Find the Highest number.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution {
public:
    int findPeakElement(vector<int>& a) 
    {
        // Code here.
        int n = a.size();
        int ans = a[n/2];
        
        int inizio = n/2;
        int fine = n-1;
        
        while (inizio <= fine)
        {
            int pivot = (inizio+fine) /2;
            if (ans <= a[pivot])
            {
                ans = a[pivot];
                inizio = pivot +1;
            }
            else fine = pivot -1;
        }
        
        return ans;
        
    }
};



//{ Driver Code Starts.
int main(){
    int T;
    cin >> T;
    while(T--)
    {
    	int n;
    	cin >> n;
    	vector<int>a(n);
    	for(int i = 0; i < n; i++)
    		cin>>a[i];
    	Solution ob;
    	int ans = ob.findPeakElement(a);
    	cout << ans << "\n";
    }
	return 0;
}


// } Driver Code Ends

File: Easy/Find the Highest number.py
#User function Template for python3
from typing import List

class Solution:
	def findPeakElement(self, a):
		# Code here
		n = len(a)
		
		if (n == 1):
		    return a[0];
		
		if (n == 2):
		    return max(a[0], a[1])
		
		start = 1
        end = n - 2
        
        if a[0] > a[start]:
            return a[0]
            
            if a[n - 1] > a[end]:
                return a[n - 1]
        
        while start <= end:
            mid = (start + end) // 2
            
            if a[mid] > a[mid - 1] and a[mid] > a[mid + 1]:
                return a[mid]
            elif a[mid-1] < a[mid]:
                start = mid + 1
            else:
                end = mid - 1
		
        return a[n-1];


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    T = int(input())
    for i in range(T):
        n = int(input())
        a = list(map(int, input().split()))
        ob = Solution()
        ans = ob.findPeakElement(a)
        print(ans)

# } Driver Code Ends